hpx/parallel/execution_policy.hpp

```diff
@@ -162,24 +163,23 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
         ///                     execution of the parallel algorithm the
         ///                     returned execution policy is used with.
         ///
-        /// \note Requires: is_executor_parameters<Parameters>::value is true
+        /// \note Requires: all parameters are executor_parameters,
+        ///                 different parameter types can't be duplicated
         ///
         /// \returns The new sequential_task_execution_policy
         ///
-        template <typename Parameters>
+        template <typename... Parameters, typename ParametersType =
+            typename hpx::parallel::executor_parameters_join<Parameters...>::type>
```

Comment:
Of course we can, it's just how I've been using it all the time. I got used to this solution, because this allows me to use it easily in specifying type before function body or in inheritance. Hence I do not need to declare this type twice, one for return type of function and then again in the function body. 

Is it considered a bad practice, bad style? Should I change it? I thought that it's better to have it as a default parameter rather than duplicate code, just like it happens in rebound_type.